runtime/src/iree/task/poller.c

```diff
@@ -420,6 +420,12 @@ static void iree_task_poller_commit_wait(iree_task_poller_t* poller,
                                          iree_time_t deadline_ns) {
   IREE_TRACE_ZONE_BEGIN(z0);
 
+  if (iree_atomic_load_int32(&poller->state, iree_memory_order_seq_cst) ==
+      IREE_TASK_POLLER_STATE_EXITING) {
+    // Thread exit requested - don't block shutdown.
+    return;
```

Comment:
you'll want to move this above the IREE_TRACE_ZONE_BEGIN or add an IREE_TRACE_ZONE_END